File: frontend/src/api/feedback.ts
import { apiClient, unwrap } from './client'

export interface FeedbackPayload {
  conversation_id?: number | null
  rating?: number | null
  sentiment?: string | null
  content?: string | null
}

export interface Feedback {
  id: number
  user_id: number | null
  conversation_id: number | null
  rating: number | null
  sentiment: string | null
  content: string | null
  created_at: string
}

export interface FeedbackAnalysis {
  sentiment: string
  satisfaction_score: number
  priority: string
  reason: string
}

export interface FeedbackAttentionItem {
  id: number
  rating: number | null
  sentiment: string
  satisfaction_score: number
  priority: string
  content: string | null
  created_at: string
}

export interface FeedbackStats {
  total: number
  average_rating: number
  average_satisfaction: number
  sentiment_counts: Record<string, number>
  priority_counts: Record<string, number>
  latest_at: string | null
  attention_items: FeedbackAttentionItem[]
}

export function submitFeedback(payload: FeedbackPayload) {
  return unwrap<Feedback>(apiClient.post('/api/feedback', payload))
}

export function listFeedback() {
  return unwrap<Feedback[]>(apiClient.get('/api/feedback'))
}

export function analyzeFeedback(payload: Pick<FeedbackPayload, 'rating' | 'content'>) {
  return unwrap<FeedbackAnalysis>(apiClient.post('/api/feedback/analyze', payload))
}

export function getFeedbackStats() {
  return unwrap<FeedbackStats>(apiClient.get('/api/feedback/stats'))
}

File: backend/app/schemas/feedback.py
from datetime import datetime

from pydantic import BaseModel, Field


class FeedbackCreate(BaseModel):
    conversation_id: int | None = None
    rating: int | None = Field(default=None, ge=1, le=5)
    sentiment: str | None = Field(default=None, max_length=32)
    content: str | None = Field(default=None, max_length=1000)


class FeedbackOut(BaseModel):
    id: int
    user_id: int | None
    conversation_id: int | None
    rating: int | None
    sentiment: str | None
    content: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class FeedbackAnalysisRequest(BaseModel):
    rating: int | None = Field(default=None, ge=1, le=5)
    content: str | None = Field(default=None, max_length=1000)


class FeedbackAnalysisOut(BaseModel):
    sentiment: str
    satisfaction_score: float
    priority: str
    reason: str


class FeedbackAttentionItem(BaseModel):
    id: int
    rating: int | None
    sentiment: str
    satisfaction_score: float
    priority: str
    content: str | None
    created_at: datetime


class FeedbackStatsOut(BaseModel):
    total: int
    average_rating: float
    average_satisfaction: float
    sentiment_counts: dict[str, int]
    priority_counts: dict[str, int]
    latest_at: datetime | None
    attention_items: list[FeedbackAttentionItem]
